sentinel5dl/__main__.py

```diff
@@ -7,30 +7,149 @@
 :url: https://emissions-api.org
 :license: MIT
 '''
-
+import argparse
+import iso8601
 import logging
+import textwrap
 
 import sentinel5dl
 from sentinel5dl import search, download
 
+PRODUCTS = (
+    'L1B_IR_SIR',
+    'L1B_IR_UVN',
+    'L1B_RA_BD1',
+    'L1B_RA_BD2',
+    'L1B_RA_BD3',
+    'L1B_RA_BD4',
+    'L1B_RA_BD5',
+    'L1B_RA_BD6',
+    'L1B_RA_BD7',
+    'L1B_RA_BD8',
+    'L2__AER_AI',
+    'L2__AER_LH',
+    'L2__CH4___',
+    'L2__CLOUD_',
+    'L2__CO___',
+    'L2__HCHO__',
+    'L2__NO2__',
+    'L2__NP_BD3',
+    'L2__NP_BD6',
+    'L2__NP_BD7',
+    'L2__O3_TCL',
+    'L2__O3___',
+    'L2__SO2__',
+)
+
+PRODUCTS_STR = textwrap.fill(', '.join(PRODUCTS),
+                             subsequent_indent='  ',
+                             initial_indent='  ')
+
+PROCESSING_LEVELS = (
+    'L1B',
+    'L2'
+)
+
+
+def is_polygon(polygon):
+    '''Validate if the supplied polygon string is in the necessary format to be
+    used as part of a WKT polygon string.
+
+    :param polygon: Polygon string
+    :return: WKT polygon
+    '''
+    values = [value.strip() for value in polygon.split(',')]
+
+    # Polygon must be at least a triangle
+    if len(values) < 4:
+        raise ValueError()
+
+    # Check if we got float pairs
+    for value in values:
+        if len([float(x) for x in value.split()]) != 2:
+            raise ValueError('Polygon values must be pairs of numbers')
+
+    # Check if we got a closed polygon
+    if values[0] != values[-1]:
+        raise ValueError()
+
+    return f'POLYGON(({polygon}))'
```

Comment:
Since we verified that `polygon` is valid, no need to construct it again from its parts. we can simply use it.